src/autoscaler/cred_helper/plugin/loader.go

```diff
@@ -0,0 +1,100 @@
+package plugin
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+
+	"code.cloudfoundry.org/app-autoscaler/src/autoscaler/cred_helper"
+	"code.cloudfoundry.org/app-autoscaler/src/autoscaler/custom-metrics-cred-helper-plugin/custom_metrics"
+	"code.cloudfoundry.org/app-autoscaler/src/autoscaler/db/sqldb"
+	"code.cloudfoundry.org/app-autoscaler/src/autoscaler/models"
+	"code.cloudfoundry.org/app-autoscaler/src/autoscaler/storedprocedure"
+	"github.com/hashicorp/go-hclog"
+
+	"code.cloudfoundry.org/app-autoscaler/src/autoscaler/db"
+	"code.cloudfoundry.org/app-autoscaler/src/autoscaler/helpers"
+
+	"github.com/hashicorp/go-plugin"
+)
+
+type CredHelperOptions string
+
+const (
+	credHelperDefault         = "default"
+	credHelperStoredProcedure = "stored_procedure"
+)
+
+type Manager struct {
+	client *plugin.Client
+}
+
+func (p *Manager) Kill() {
+	if p.client != nil {
+		p.client.Kill()
+	}
+}
+
+func (p *Manager) LoadCredentialsImplementation(dbConfigs map[db.Name]db.DatabaseConfig, loggingConfig helpers.LoggingConfig, credHelperPlugin string, storedProcedureConfig *models.StoredProcedureConfig) (cred_helper.Credentials, error) {
+	if credHelperPlugin == "" {
+		return nil, errors.New("credHelperPlugin is not configured")
+	}
+
+	switch credHelperPlugin {
```

Comment:
Id consider putting this outside of the load since it is supposed to be loading the plugin not newing different instances